### DISCUSSION TITLE
How to reset mongodb before test

### DISCUSSION THREAD

**Discussion by crapthings** [fastify/fastify#4121] [Q&A]
i try to clean database with before, but how

```js
require('dotenv').config()

const { before, test } = require('tap')

const Fastify = require('../src')

let fastify

before(async function () {
  console.log(1)
  fastify = Fastify()

  // await fastify.mongo.db.collection('users').remove({})
})

test(`post /api/register`, async (t) => {
  const fastify = Fastify()

  t.teardown(() => fastify.close())

  const resp = await fastify.inject()
    .post('/api/register')
    .payload({ username: 'test', password: 'test' })

  t.type(resp.statusCode)
})
```

**Comment by Eomm** [comment]
You don't have access to your mongo db instance because it is not registered in the root scope (the `fastify` var from `../src`).

You need to connect to your db by using the nodejs MongoDB driver or try a plugin to get the connection even when it is hidden: https://github.com/Eomm/fastify-explorer

**Comment by mcollina** [comment]
A long time ago I wrote https://github.com/mcollina/mongo-clean. I don't know if it still works.
